3party/opening_hours/opening_hours_integration_tests/opening_hours_integration_tests.cpp

```diff
@@ -0,0 +1,215 @@
+#include "parse_opening_hours.hpp"
+#include "rules_evaluation.hpp"
+
+#define BOOST_TEST_MODULE OpeningHoursIntegration
+
+#include <algorithm>
+#include <ctime>
+#include <map>
+
+#include <boost/algorithm/string/case_conv.hpp>
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/test/included/unit_test.hpp>
+
+namespace
+{
+template <typename T>
+std::string ToString(T const & t)
+{
+  std::stringstream sstr;
+  sstr << t;
+  return sstr.str();
+}
+
+template <typename T>
+bool HasPeriod(std::vector<T> const & v)
+{
+  auto const hasPeriod = [](T const & t) { return t.HasPeriod(); };
+  return std::any_of(begin(v), end(v), hasPeriod);
+}
+
+template <typename T>
+bool HasPlus(std::vector<T> const & v)
+{
+  auto const hasPlus = [](T const & t) { return t.HasPlus(); };
+  return std::any_of(begin(v), end(v), hasPlus);
+}
+
+bool HasEHours(osmoh::TTimespans const & spans)
+{
+  auto const hasEHours = [](osmoh::Timespan const & s) -> bool {
+    if (!s.HasEnd())
+      return false;
+    return s.GetEnd().GetMinutes() + s.GetEnd().GetHours() > 24 * std::chrono::minutes(60);
+  };
+  return std::any_of(begin(spans), end(spans), hasEHours);
+}
+
+bool HasOffset(osmoh::TMonthdayRanges const & mr)
+{
+  auto const hasOffset = [](osmoh::MonthdayRange const & md) {
+    return
```

Comment:
Open brace should be on the next line and return value should be on the same line as return keyword.